packages/compass-editor/src/codemirror/query-history-autocompleter.ts

```diff
@@ -92,7 +96,34 @@ function createInfo(query: SavedQuery): {
 
     const code = document.createElement('pre');
     code.className = queryCodeStyles;
-    if (formattedQuery) code.textContent = formattedQuery;
+
+    function emit(text: string, classes: string | null) {
+      const node = document.createTextNode(text);
+      if (classes) {
+        const span = document.createElement('span');
+        span.appendChild(node);
+        span.className = classes;
+        code.appendChild(span);
+      } else {
+        code.appendChild(node);
+      }
+    }
+
+    function emitBreak() {
+      code.appendChild(document.createTextNode('\n'));
+    }
+
+    if (formattedQuery) {
+      highlightCode(
+        formattedQuery,
+        languages['javascript-expression']().language.parser.parse(
+          formattedQuery
+        ),
```

Comment:
A slight performance improvement (I'm thinking it could factor in if someone is scrolling through a lot of query history items), could we define this parser once so we don't rebuild this language support object each time? It's probably not too costly, but I think it doesn't hurt to move it to a constant variable outside of this function.

Reply:
How much of it do I move out to a constant variable? Including `parser` or just up to` languages['javascript-expression']().language`?